### ISSUE TITLE
Tokenization Bug With Bigrams in Exception List

### ISSUE THREAD

**Issue by KatieGoz** [JohnSnowLabs/spark-nlp#983]
## Description
Bigrams in tokenization exception list appear to be tokenized with subsequent punctuation. (Expected behavior is that the punctuation would be tokenized separately.) Also, when sentence is input into tokenizer (and bigram exception is present) the bigram token is sometimes dropped altogether. 

## Expected Behavior
Bigram exceptions should be tokenized without punctuation, and should not be dropped from output regardless of whether document or sentence is input to tokenizer.

## Current Behavior
For example, if the bigram exception in the tokenizer is "New York" and the input text is "My friend moved to New York. She likes it. Frank visited New York, and didn't like it." Then tokenized output includes "New York." and "New York," when document is the input for tokenizer. When sentence is the input for tokenizer, "New York." is output and "New York," is dropped from output.

## Steps to Reproduce
See: https://github.com/kgoz12/JSL_Bugs

1.
2.
3.
4.

## Context
I was piping tokenized text into context spell checker. Spell checker replacements will not contain punctuation, causing improperly punctuated sentence output.

## Your Environment

* Spark NLP version: '2.5.4'
* Apache NLP version:
* Java version (java -version): 8
* Setup and installation (Pypi, Conda, Maven, etc.):
* Operating System and version: MacOS
* Link to your project (if any): https://github.com/kgoz12/JSL_Bugs

**Comment by dkincaid** [comment]
I have come across a very similar problem. I'm using R, so my example code probably won't help you much. In my case though the LightPipeline and the PIpeline are both having the problem. I'll try to explain what I'm seeing. I am creating a Spark data frame with one column, `text` that contains one row with the following text:

`No cough/sneezing noted in 1/2 of the cases. No other findings. Eyes: [X] Normal [_] Did not examine [_] Abnormal`

then I create a Spark NLP pipeline consisting of document assembler, sentence detector and tokenizer. The document assembler and sentence detector use all default values. For the tokenizer I am specifying `"[_]", "[X]"` as exceptions using `setExceptions`. I next fit the pipeline and also create a LightPipeline from it. 

If I specify the `sentence` as the input column for the tokenizer, the tokens "[_]" and "[X]" from the input text are not in the list of the results in either the Pipeline or LightPipeline cases. If I specify `document` as the input column for the tokenizer then these tokens are present in the results.

Here is the code for the example in case it does help:
```
document_assembler <- nlp_document_assembler(sc, input_col = "text", output_col = "document")
sentence_detector <- nlp_sentence_detector(sc, input_cols = c("document"), output_col = "sentence")
tokenizer <- nlp_tokenizer(sc, input_cols = c("sentence"), output_col = "token", exceptions = c("[X]", "[_]"))

empty_df <- sdf_copy_to(sc, data.frame(text = ""))

pipeline <- ml_pipeline(document_assembler, sentence_detector, tokenizer) %>% 
   ml_fit(empty_df)

lp <- nlp_light_pipeline(pipeline)

text_df <- sdf_copy_to(sc, data.frame(text = c("No cough/sneezing noted in 1/2 of the cases.No other findings. Eyes: [X] Normal [_] Did not examine [_] Abnormal")))

transform_result <- ml_transform(pipeline, text_df)

transform_result %>% 
   mutate(token_exploded = explode(token),
          token = token_exploded.result)

nlp_annotate(lp, "No cough/sneezing noted in 1/2 of the cases.No other findings. Eyes: [X] Normal [_] Did not examine [_] Abnormal")
```

**Comment by KatieGoz** [comment]
Thank you for the explanation Alberto! I really appreciate your looking into this!

**Comment by maziyarpanahi** [comment]
This should be fixed in 2.6.0, please feel free to reopen it if it still exists.
